Yoast/wordpress-seo
# Cherry pick modal changes from feature metabox modal

## Context


* The special projects team is also working on a modal. They should be able to use the same one as plugin components. So we are extracting the general modal changes from the feature branch to `trunk`.

See https://github.com/Yoast/wordpress-seo/pull/15815 for context

## Summary


This PR can be summarized in the following changelog entry:

* We now use the wordpress modal rather than a in-house one.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Verify whether the Search Editor and the `Add synonyms` and `Add related keyphrase` (under SEO analysis) modals still work.
(this one: 
![image](https://user-images.githubusercontent.com/26220788/90485910-7bd20100-e138-11ea-9673-8870e9ab29ad.png) and these ones:
![image](https://user-images.githubusercontent.com/26220788/90486552-6b6e5600-e139-11ea-9248-1e2e3708ae38.png)
 )

* Verify that there are no errors when building and browsing with Free and Premium.

## UI changes

## Documentation

## Quality assurance


Fixes N/A
